src/coreComponents/physicsSolvers/multiphysics/SinglePhaseReservoirAndWells.cpp

```diff
@@ -33,46 +32,102 @@ namespace geosx
 using namespace dataRepository;
 using namespace constitutive;
 
-SinglePhaseReservoir::SinglePhaseReservoir( const string & name,
-                                            Group * const parent ):
-  ReservoirSolverBase( name, parent )
+namespace
 {
-  m_linearSolverParameters.get().mgr.strategy = LinearSolverParameters::MGR::StrategyType::singlePhaseReservoirFVM;
+
+template< typename SINGLEPHASE_RESERVOIR_SOLVER > class
+  SinglePhaseCatalogNames {};
+
+template<> class SinglePhaseCatalogNames< SinglePhaseFVM< SinglePhaseBase > >
+{
+public:
+  static string name() { return "SinglePhaseReservoirFVM"; }
+};
+template<> class SinglePhaseCatalogNames< SinglePhaseHybridFVM >
+{
+public:
+  static string name() { return "SinglePhaseReservoirHybridFVM"; }
+};
+
 }
 
-SinglePhaseReservoir::~SinglePhaseReservoir()
+// provide a definition for catalogName()
+template< typename SINGLEPHASE_RESERVOIR_SOLVER >
+string
+SinglePhaseReservoirAndWells< SINGLEPHASE_RESERVOIR_SOLVER >::
+catalogName()
+{
+  return SinglePhaseCatalogNames< SINGLEPHASE_RESERVOIR_SOLVER >::name();
+}
+
+template< typename SINGLEPHASE_RESERVOIR_SOLVER >
+SinglePhaseReservoirAndWells< SINGLEPHASE_RESERVOIR_SOLVER >::
+SinglePhaseReservoirAndWells( const string & name,
+                              Group * const parent )
+  : Base( name, parent )
+{}
+
+template< typename SINGLEPHASE_RESERVOIR_SOLVER >
+SinglePhaseReservoirAndWells< SINGLEPHASE_RESERVOIR_SOLVER >::
+~SinglePhaseReservoirAndWells()
 {}
 
-void SinglePhaseReservoir::initializePostInitialConditionsPreSubGroups()
+template< typename SINGLEPHASE_RESERVOIR_SOLVER >
+void
```

Comment:
I must confess I'm a bit surprise to see those templates in a `cpp` file (same for some other class `CoupledReservoirAndWellsBase.cpp` above.)

Reply:
Yes, you are right, I am going to work on that now

Reply:
I removed the templates from the implementations in the `CoupledReservoirAndWells.cpp` file, but I left them in `SinglePhaseReservoirAndWells.cpp` and `CompositionalMultiphaseReservoirAndWells.cpp`. I left the templates in these files because the derived classes `SinglePhaseFVM.cpp` and `CO2BrineFluid.cpp` follow the same pattern (i.e., they have templates), but if you think they should be removed in `SinglePhaseReservoirAndWells.cpp` and `CompositionalMultiphaseReservoirAndWells.cpp` as well, let me know.